hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -479,14 +489,18 @@ def all(cls) -> List["HealthCheck"]:
     testing."""
 
     return_value = 5
-    """Checks if your tests return a non-None value (which will be ignored and
+    """Deprecated, now a strict error.
+    
+    Checks if your tests return a non-None value (which will be ignored and
     is unlikely to do what you want)."""
```

Comment:
```suggestion
    """Deprecated; the check is now an unconditional error."""
```

and likewise below.

Reply:
I assumed that it would be helpful to keep the old description for previous uses, should I take that out as well?

Reply:
Hmm, good point 🤔   I'll push a tighter phrasing but keep some description of what it does for now.